src/pages/groupComparison/GenericAssayEnrichments.tsx

```diff
@@ -0,0 +1,129 @@
+            profileMap,
+            this.props.genericAssayType
+        );
+    }
+
+    readonly tabUI = MakeEnrichmentsTabUI(
+        () => this.props.store,
+        () => this.enrichmentsUI,
+        this.props.genericAssayType,
+        true,
+        true,
+        false
+    );
+
+    readonly enrichmentsUI = MakeMobxView({
+        await: () => {
+            const ret: any[] = [
+                this.props.store
+                    .genericAssayEnrichmentDataGroupByGenericAssayType,
+                this.props.store
+                    .selectedGenericAssayEnrichmentProfileMapGroupByGenericAssayType,
+                this.props.store
+                    .genericAssayEnrichmentAnalysisGroupsGroupByGenericAssayType,
+                this.props.store.studies,
+            ];
+            if (
+                this.props.store
+                    .genericAssayEnrichmentDataGroupByGenericAssayType
+                    .isComplete &&
+                this.props.store
+                    .genericAssayEnrichmentDataGroupByGenericAssayType.result![
+                    this.props.genericAssayType
+                ]
+            ) {
+                ret.push(
+                    this.props.store
+                        .genericAssayEnrichmentDataGroupByGenericAssayType
+                        .result![this.props.genericAssayType]
+                );
+            }
+            return ret;
+        },
+        render: () => {
+            // since generic assay enrichments tab is enabled only for one study, selectedGenericAssayEnrichmentProfileMap
+            // would contain only one key.
+            const studyIds = Object.keys(
+                this.props.store
+                    .selectedGenericAssayEnrichmentProfileMapGroupByGenericAssayType
+                    .result![this.props.genericAssayType]
+            );
+            const selectedProfile = this.props.store
+                .selectedGenericAssayEnrichmentProfileMapGroupByGenericAssayType
+                .result![this.props.genericAssayType][studyIds[0]];
+            return (
+                <div data-test="GroupComparisonGenericAssayEnrichments">
+                    <GenericAssayEnrichmentsDataSetDropdown
+                        dataSets={
+                            this.props.store
+                                .genericAssayEnrichmentProfilesGroupByGenericAssayType
+                                .result![this.props.genericAssayType]
+                        }
+                        onChange={this.onChangeProfile}
+                        selectedProfileByStudyId={
+                            this.props.store
+                                .selectedGenericAssayEnrichmentProfileMapGroupByGenericAssayType
+                                .result![this.props.genericAssayType]
+                        }
+                        alwaysShow={true}
+                        studies={this.props.store.studies.result!}
+                    />
```

Comment:
> @dippindots looks like profile type(dataset) is appended to `GENERIC_ASSAY`, any specific reason? I think it is better to have it in its own field.
> 
> @alisman @jjgao @adamabeshouse any thoughts?

@kalletlak Do you mean this `GenericAssayEnrichmentsDataSetDropdown`? This data set component is specific to `Generic Assay` since I cannot pass a Promise as the source of `dataSets`. I need to know which `generic assay type` is showing. That's why I created this new component.

Reply:
It is about `comparison_subtab` param in url. It is a join of `GENERIC_ASSAY` and dataset, example `GENERIC_ASSAY_MICROBIOME_SIGNATURE`. Instead we could have `comparison_subtab` as `generic_assay` and `MICROBIOME_SIGNATURE` as new parameter `dataset`. It would be nice if we standardize this even for other comparison subtabs.

Reply:
@kalletlak I see, it's a good idea to separate the datasets as a new parameter. However I don't think I should leave `Generic Assay` as the `comparison_subtab` id since we will create multiple `comparison_subtabs` (one for each `generic assay type`), and then can select `dataset` from each subtabs.
https://github.com/cBioPortal/cbioportal-frontend/pull/3351/files#diff-d27fe25dc0b90c52a1e119c2a588de84R247-R276